docs/ccv1_image_security_design.md

```diff
@@ -0,0 +1,509 @@
+For example, if the owner need to completely disable the container image signature verification function 
+(for example, the owner thinks that encryption alone is enough to meet his security requirements), 
+policy file can be:
+
+```json
+{
+  "default": [
+    {
+      "type": "insecureAcceptAnything"
+    }
+  ],
+}
+```
+
+**Warning**: this configuration will make the pod accept all unsigned container images. 
+
+If the owner want to unconditionally accept an image from a specific registry, 
+he just need to change the policy of the member matching the registry under the corresponding scope to `insecureAcceptAnything`.
+
+## Signatures verification
+
+### Get signatures
+
+If the image is pulled from the remote registry, 
+first try to request signatures with the registry's `X-Registry-Supports-Signatures` API extension. 
+If the image is pulled from the local rootfs or the registry does not support the extension, 
+read the sigstore configuration file to get the sigstore URL of the image 
+and then get signatures from the sigstore.
+
+The sigstore configuration file is specified as all `.yaml` files in `registries.d` directory. 
+The entries in the sigstore configuration file will claim the sigstore URL of the container image
+(the sigstore can be a webserver, local directory or etc.).
+
+The spec of sigstore configuration file is defined in 
+[contianers/image registry.d](https://github.com/containers/image/blob/main/docs/containers-registries.d.5.md).
+
+The sample sigstore configuration file matching the sample policy.json file in the above is as follows:
+
+```yaml
+default:
+    sigstore: file:///var/lib/containers/sigstore
+docker:
+    docker.io/my_private_registry:
+        sigstore: https://my-sigstore.com/example/sigstore
+    registry.access.redhat.com:
+        sigstore: https://access.redhat.com/webassets/docker/content/sigstore
+```
+
+### Verification steps
+
+A single signature's verification action is divided into the following steps:
+
+1. Parse the signature claim file and separate the JSON payload and cryptographic signature.
+
+2. Read the policy.json's `signedBy` policy, get the public key used to verify the signature from the `keyPath` or `keyData` field.
+
+3. Use the public key to verify the cryptographic signature in the signature claim file.
+
+4. Compare whether:
+   - the manifest digest in the JSON payload is consistent with that of the container image
+   - the docker reference in the JSON payload is consistent with that of the container image.
+
+## Image layer decryption
+
+After verifying the container image signature, image-rs will start the actual image pulling. 
+for each encrypted image layer, 
+image-rs calls the container image cryptography operation APIs provided by ocicrypt-rs 
+to decrypt the encrypted image layer. ocicrypt-rs perform the following steps:
+
+1. Read the layer's two annotations fields in the image manifest. 
```

Comment:
Hmm, again this idea that there is a per-layer annotation. So I guess that must exist somewhere. A link would help.

Reply:
Annotation here refers to [annotations.md](https://github.com/opencontainers/image-spec/blob/main/annotations.md), annotations can be flexibly placed in many locations in the manifest, such as:

```json
{
    "schemaVersion": 2,
    "config": {
        ...
    },
    "layers": [
        {
            "mediaType": "application/vnd.oci.image.layer.v1.tar+gzip,
            "digest": "sha256:f69ae40...",
            "size": 204812,
        }
    ]
   "annotations": {
                ...
    }
}
```

Or: 

```json
{
    "schemaVersion": 2,
    "config": {
        ...
    },
    "layers": [
        {
            "mediaType": "application/vnd.oci.image.layer.v1.tar+gzip+encrypted",
            "digest": "sha256:f69ae40...",
            "size": 204812,
            "annotations": {
                ...
            }
        }
    ]
}
```

The latter is what we call pre-layer annotations.